src/InstallationValidator.Core/Services/BatchOutputFileComparer.cs

```diff
@@ -38,16 +38,28 @@ public Task<OutputFileComparisonResult> Compare(string fileName, ComparisonSetti
             outputFileComparision.AbsTol = simulation1.Simulation.AbsTol;
             outputFileComparision.RelTol = simulation1.Simulation.RelTol;
 
+
+            var outputComparisonResults = new List<OutputComparisonResult>();
             foreach (var outputValue in simulation1.Simulation.OutputValues.Where(p => simulation2.HasOutput(p.Path)))
             {
                token.ThrowIfCancellationRequested();
-               outputFileComparision.AddOutputComparison(compareOutputs(simulation1, simulation2, outputValue, simulation2.OutputByPath(outputValue.Path),comparisonSettings));
+               outputComparisonResults.Add(compareOutputs(simulation1, simulation2, outputValue, simulation2.OutputByPath(outputValue.Path), comparisonSettings));
             }
 
+            outputFileComparision.AddOutputComparisons(selectOutputComparisonFrom(outputComparisonResults, comparisonSettings));
+           
             return outputFileComparision;
          }, token);
       }
 
+      private IEnumerable<OutputComparisonResult> selectOutputComparisonFrom(IReadOnlyList<OutputComparisonResult> outputComparisonResults, ComparisonSettings comparisonSettings)
+      {
+         if (!comparisonSettings.NumberOfCurves.HasValue)
+            return outputComparisonResults;
+
+         return outputComparisonResults.OrderByDescending(x => x.Deviation).Take(comparisonSettings.NumberOfCurves.Value);
```

Comment:
This is the difference. If the number of curves is set, order by deviation and only take number of curves

Reply:
If the user starts OutputsComparer via UI and deletes Number Of Curves in the input text box: will it be set to null, so that all curves will be exported?